github/src/wizard/GitHubRepositoryListStep.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import { QuickPickItem, l10n } from "vscode";
+import type { GitHubContext } from "../GitHubContext";
+import { loadMoreQp } from "../constants";
+import { GetOrgReposReqParams, GetUserReposReqParams, getRepositoriesByOrg, getRepositoriesByUser } from "../wrappers/getRepositories";
+
+type RepositoryData = {
+    owner: string;
+    repo: string;
+    url: string;
+}
+
+export class GitHubRepositoryListStep extends AzureWizardPromptStep<GitHubContext> {
+    private picks: IAzureQuickPickItem<RepositoryData>[];
+
+    public async prompt(context: GitHubContext): Promise<void> {
+        // We always want fresh picks before prompting in case the user has pressed the back button
+        this.picks = [];
+
+        const placeHolder: string = l10n.t('Select a repository');
+
+        let page: number = 0;
+        let repositoryData: RepositoryData | undefined;
+        while (!repositoryData) {
+            page++;
+            repositoryData = (await context.ui.showQuickPick(this.getPicks(context, page), { placeHolder })).data;
+        }
+
+        context.gitHubRepository = repositoryData.repo;
+        context.gitHubRepositoryOwner = repositoryData.owner;
+        context.gitHubRepositoryUrl = repositoryData.url;
+
+        context.valuesToMask.push(context.gitHubRepository);
+        context.valuesToMask.push(context.gitHubRepositoryOwner);
+        context.valuesToMask.push(context.gitHubRepositoryUrl);
+    }
+
+    public shouldPrompt(context: GitHubContext): boolean {
+        return !context.gitHubRepository || !context.gitHubRepositoryOwner;
+    }
+
+    private async getPicks(context: GitHubContext, page: number): Promise<IAzureQuickPickItem<RepositoryData | undefined>[]> {
+        const perPage: number = 50;
+        const userRepoParams: GetUserReposReqParams = {
+            affiliation: 'owner',
+            visibility: 'all',
+            per_page: perPage,
+            page
+        };
+        const orgRepoParams: GetOrgReposReqParams = {
+            // Doesn't matter if context.gitHubOrg is undefined; that just means we will end up calling 'getRepositoriesByUser' instead
+            org: context.gitHubOrg ?? '',
+            type: 'all',
+            per_page: perPage,
+            page
+        };
+
+        this.picks.push(
+            ...(context.gitHubOrg ?
+                (await getRepositoriesByOrg(context, orgRepoParams)).map((repo) => { return { label: repo.name, data: { owner: repo.owner.login, repo: repo.name, url: repo.html_url } } }) :
```

Comment:
minor nit: Kind of bothers me that this is copy pasta. Maybe we can just do something like:

`const repos = context.gitHubOrg ? ...`

And then do:

`this.picks.push(...repos.map(repo => ))...`

Reply:
What do you mean by copy pasta? And sure, I'll make that change

Reply:
Copy pasta = copy and paste = `.map((repo) => { return { label: repo.name, data: { owner: repo.owner.login, repo: repo.name, url: repo.html_url } } })` being used twice.

Reply:
Oooooh, I see what you mean now.

Reply:
I think I remember why I had it that way, the type inference when you do the map with the interchangeable repo sources seems to stump Typescript into typecasting to an `any`.  I don't know the actual array element types to manually cast it to, so it was a lot easier to just keep the maps strict like that to force the proper type inference

Reply:
Added a comment to better explain the old way of doing it.  Let me know if you prefer a different way still